File: tests/conformance/sandbox-tools/fetch_blocked.rb
tool description: "Fetch a blocked domain",
     input: {},
     permissions: { network: ["localhost"] }

execute do |args, ctx|
  if ZeroMcp::Sandbox.check_network_access(ctx.tool_name, "evil.test", ctx.permissions)
    { "blocked" => false }
  else
    { "blocked" => true, "domain" => "evil.test" }
  end
end

File: tests/conformance/sandbox-tools/fetch_blocked.php
<?php

return [
    'description' => 'Fetch a blocked domain',
    'input' => [],
    'permissions' => ['network' => ['localhost']],
    'execute' => function ($args, $ctx) {
        if (\ZeroMcp\Sandbox::checkNetworkAccess($ctx->toolName, 'evil.test', $ctx->permissions)) {
            return ['blocked' => false];
        }
        return ['blocked' => true, 'domain' => 'evil.test'];
    },
];
